### PR TITLE
Modifies promote action to handle tracks

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

Modifies promote action to handle tracks

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by Gmerold — APPROVED** [review]

File: .github/workflows/promote.yaml [hunk 1]
```diff
@@ -13,8 +13,9 @@ on:
 
 jobs:
   promote:
+    if: contains([1.6, main], github.ref_name)
     name: Promote Charm
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (gruyaume) [inline .github/workflows/promote.yaml:18]: Why not use 22.04?

File: .github/workflows/promote.yaml [hunk 2]
```diff
@@ -32,11 +33,19 @@ jobs:
             echo "promote-from=candidate" >> ${GITHUB_ENV}
             echo "promote-to=stable" >> ${GITHUB_ENV}
           fi
-      - name: Promote Charm
+      - name: Promote charm to latest
+        if: github.ref_name==main
         uses: canonical/charming-actions/release-charm@2.0.0
```
Comment (gruyaume) [inline .github/workflows/promote.yaml:38]: We should use `2.2.2`

File: .github/workflows/promote.yaml [hunk 3]
```diff
@@ -32,11 +33,19 @@ jobs:
...
         with:
           credentials: ${{ secrets.CHARMCRAFT_AUTH }}
           github-token: ${{ secrets.GITHUB_TOKEN }}
           destination-channel: latest/${{ env.promote-to }}
           origin-channel: latest/${{ env.promote-from }}
-          charmcraft-channel: latest/stable
+      - name: Promote charm to release track
+        if: github.ref_name!=main
+        uses: canonical/charming-actions/release-charm@2.0.0
```
Comment (gruyaume) [inline .github/workflows/promote.yaml:46]: We should use `2.2.2`

File: .github/workflows/promote.yaml (hunk 1 above)
Comment (saltiyazan) [reply .github/workflows/promote.yaml]: I wanted to make it in sync with the other workflows.
But now I relealised that there is no need to to fix the version in this specific workflow file.
Keeping `ubuntu-latest`.

File: .github/workflows/promote.yaml (hunk 2 above)
Comment (saltiyazan) [reply .github/workflows/promote.yaml]: Right, ✅

File: .github/workflows/promote.yaml (hunk 3 above)
Comment (saltiyazan) [reply .github/workflows/promote.yaml]: ✅

**Review by gruyaume — APPROVED** [review]

**Pull request merged**